imap/imap.go

```diff
@@ -0,0 +1,344 @@
+package imap
+
+// Functionality taken from https://github.com/jprobinson/eazye
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"strconv"
+	"time"
+
+	log "github.com/gophish/gophish/logger"
+	"github.com/gophish/gophish/models"
+	"github.com/jordan-wright/email"
+	"github.com/mxk/go-imap/imap"
+)
+
+// Email Email struct with IMAP UID
+type Email struct {
+	UID uint32 `json:"uid"`
+	*email.Email
+}
+
+// MailboxInfo holds onto the credentials and other information
+// needed for connecting to an IMAP server.
+type MailboxInfo struct {
+	Host   string
+	TLS    bool
+	User   string
+	Pwd    string
+	Folder string
+	// Read only mode, false (original logic) if not initialized
+	ReadOnly bool
+}
+
+// GetAll will pull all emails from the email folder and return them as a list.
+func GetAll(info MailboxInfo, markAsRead, delete bool) ([]Email, error) {
```

Comment:
What do you think about making all the functions here that have `MailboxInfo` as their first argument instead be functions off a `*MailboxInfo`, eg:

```go
func (mbox *MailboxInfo) GetAll(markAsRead, delete bool) ([]Email, error) {}
```

It seems like that might be a bit cleaner? Additionally, we could rename `MailboxInfo` to just `Mailbox`, since it seems to logically represent a mailbox. I'm all for simpler struct names where possible!

Both of these are 100% your call- just suggestions from me.

Reply:
Ah yes much neater this way, thanks! Done.